### PR TITLE
fix: throw during fetch & return empty profile

### PR DESCRIPTION
**Pull request by lsenta**
- [x] _linkProfile => preserve behavior
- [x] _publishRootStore => preserve behavior
- [x] getRootStoreAddress => rethrow
- [x] listSpaces => catch and return an empty array
- [x] getSpace => catch and return an empty object
- [x] getProfiles => rethrow (it's a list of profile so a "global" 404 error should be raised as an error).
- [x] getProfile => empty profile

### CODE REVIEW

**Comment by lsenta** [comment]
@oed I modified the fetch function so that it throws when we get a response that has an error status (not 2xx). 

There are a few other functions that are using fetch. I guess we want a consistent behavior for the function that are part of the public API and we want to throw when it's an internal error, so:

- `_linkProfile` => rethrow
- `_publishRootStore` => rethrow
- `getRootStoreAddress` => rethrow
- `listSpaces` => catch and return an empty array
- `getSpace` => catch and return an empty object
- `getProfiles` => rethrow (it's a list of profile so a "global" 404 error should be raised as an error).

File: src/api.js [hunk 1]
```diff
@@ -39,14 +39,12 @@ async function getSpace (address, name, serverUrl = PROFILE_SERVER_URL) {
...
-    } catch (err) {
-      reject(err)
-    }
-  })
+  try {
+    // we await explicitly here to make sure the error is catch'd in the correct scope
+    return await utils.fetchJson(serverUrl + '/profile?address=' + encodeURIComponent(address))
+  } catch (err) {
+    return {} // empty profile
+  }
```
Comment (oed) [inline src/api.js]: Heh, not sure why there was a promise here before 😅

**Comment by lsenta** [comment]
> Looks good. My main concern is with two of the methods you mentioned.
> `_linkProfile` and `_publishRootStore` Both of these should catch the error and fail silently, as that's the behavior they had before, and we dont wan't these methods failing stoping the user from interacting with their 3box.

I'm on the "fail fast, fail hard" side, if the function is critical we should iron it out. I'll keep the previous behavior (fail silently), could we plan to add some error logging ala sentry.io?

**Comment by oed** [comment]
I'm not familiar with sentry.io :) 
The main reason I'm not so worried about this is that I think we can remove these two requests when we fold the address-server into the pinning-node!

**Comment by lsenta** [comment]
@oed The code is ready for review (I still can't make the test run locally, but looks fine on circleci).

**Pull request merged**